worker/caasunitprovisioner/application_worker.go

```diff
@@ -132,17 +128,7 @@ func (aw *applicationWorker) loop() error {
 		// We must handle any processing due to application being removed prior
 		// to shutdown so that we don't leave stuff running in the cloud.
 		case <-aw.appRemoved:
-			if !aw.jujuManagedUnits {
-				continue
-			}
-			// Application has been removed, ensure all units are stopped
-			// before this worker is killed.
-			for unitId, w := range unitWorkers {
-				if err := aw.containerBroker.DeleteUnit(unitId); err != nil {
-					logger.Errorf("error deleting unit %v of removed application: %v", unitId, err)
-				}
-				worker.Stop(w)
-			}
+			continue
```

Comment:
I think the appRemoved channel can be deleted.

Reply:
yes, agreed. remove it now.